persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/PersistentEntityActor.scala

```diff
@@ -51,13 +64,19 @@ private[lagom] class PersistentEntityActor[C, E, S](
   snapshotAfter:             Int,
   passivateAfterIdleTimeout: FiniteDuration
 ) extends PersistentActor {
+
+  import PersistentEntityActor.EntityIdSeparator
   private val log = Logger(this.getClass)
 
   private val entityId: String = id.getOrElse(
     URLDecoder.decode(self.path.name, ByteString.UTF_8)
   )
+  require(
+    !persistenceIdPrefix.contains(EntityIdSeparator),
```

Comment:
should we use something more exotic as separator, so that `-` still can be used in the entity id?